ecosystem/sep-0032.md

```diff
@@ -0,0 +1,147 @@
+assets that they do not have prior knowledge.
+
+Wallets that use the home domain of an issuer addres can display user-friendly
+names but only if they already know the complete issuer address. Also, the home
+domain field of an issuer address does not provide confidence that the domain
+is related to the issuer of the asset because any Stellar account holder can
+set any domain as their home domain.
+
+Outside of these two approaches users are left referring to assets by their
+fully qualified names using their code and issuer address in forms such as
+`{asset-code}:{asset-issuer}`, e.g.
+`USD:GBAFEDJ6BNM72RXA45CRLFCAQFHRZUKPNK74SNCPFWHI2GMMGA6QAJME`.
+
+## Abstract
+
+This proposal defines a format for specifying memorable addresses that contains
+a domain (e.g. `example.org`) instead of a Stellar address for the issuing
+account. The domain hosts a `stellar.toml` containing information about the
+asset which is already a common practice of asset issuers. A client parses the
+`stellar.toml` of the domain in the asset address and retrieves from the file
+the issuer address for the asset.
+
+Reverse verification of an asset issuer address and domain can be performed by
+comparing the home domain of the issuer address with the issuer address in the
+domains `stellar.toml` file to confirm that the domain and issuer both claim to
+issue the asset.
+
+## Specification
+
+### Asset Address Format
+
+This proposal defines a format for specifying memorable addresses that resolve
+to the asset code and asset issuer and can be used to confidently reference an
+asset.
+
+The format is:
+
+```
+{asset-code}:{asset-domain}
+```
+
+Where:
+- `{asset-code}` is the 1-12 alphanumeric [Stellar asset code].
+- `{asset-domain}` the domain name that is hosting a `stellar.toml` file that
+  contains a [`CURRENCIES`] section for the asset, as defined by [SEP-1].
+
+Example:
+
+```
+USD:example.com
+```
+
+### Resolving an Asset Address
+
+The process of resolving an asset address is the process of finding the asset
+issuing `G...` address given an asset-code or asset-domain in the asset address
+format.
+
+The flow to resolve an asset is as follows:
+- The client receives an address address in the format
+  `{asset-code}:{asset-domain}`.
+- The client splits the asset address on `:` into two parts, the asset code and
+  the asset domain.
+- The client retrieves the `stellar.toml` for the domain located at
+  `https://{asset-domain}/.well-known/stellar.toml`.
+- The client parses the `stellar.toml` and finds a `CURRENCIES` section with
+  `code` equal to `{asset-code}`.
+- The client retrieves any currency.toml files as reference by the
+  `stellar.toml` as required.
+- If none is found, the client finds a `CURRENCIES` section with
+  `code_template` that matches `{asset-code}`.
+- If none is found, the client fails to resolve the asset address.
+- If one is found, the client continues to resolve the asset's issuer address.
```

Comment:
@leighmcculloch by reading SEP001, it seems possible to have two assets with the same `CODE`

Reply:
Thanks, that's a good point. I don't think there's anyway to support assets that are defined multiple times unless we say that the first currency defined with an asset code is the one resolved to, or we add a field that signals which is selected by the address address. Both of these options complicate the issue. I'd be interested to know how many meaningful asset issuers are doing this, I might spend sometime using the BigQuery data set and some scripts to see if I can figure out how many issuers do this. I also think this is a practice that should probably be discouraged though since an issuing organization reusing the same code is confusing for consumers in general. Asset addresses are geared towards consumers so we could just say it doesn't provide a capability to reference multiple when they are multiple. I added a section about that to the proposal for now in 2e62e03.